sklearn/decomposition/nmf.py

```diff
@@ -55,55 +58,117 @@ def check_non_negative(X, whom):
         raise ValueError("Negative values in data passed to %s" % whom)
 
 
-def _initialize_nmf(X, n_components, variant=None, eps=1e-6,
+def _check_init(A, shape, whom):
+    A = check_array(A)
+    if np.shape(A) != shape:
+        raise ValueError('Array with wrong shape passed to %s. Expected %s, '
+                         'but got %s ' % (whom, shape, np.shape(A)))
+    check_non_negative(A, whom)
+    if np.max(A) == 0:
```

Comment:
With a full col/row of zeros in W or H, WtW or HHt will have a zero in its diagonal, which is a problem since we will divide by zero.
To avoid that, I added a small L2 regularization (1e-15), which avoid this problem.